pkg/app/mutate_app.go

```diff
@@ -77,12 +78,12 @@ func (m *Mutator) Mutate(request *v1beta1.AdmissionRequest) ([]mutator.PatchOper
 
 	// We check the deletion timestamp because app CRs may be deleted by
 	// deleting the namespace they belong to.
-	if !appNewCR.DeletionTimestamp.IsZero() {
+	if request.Operation == v1beta1.Update && !appNewCR.DeletionTimestamp.IsZero() {
```

Comment:
I made this check explicit so there to hopefully reduce confusion.

Reply:
That's a good shout, thanks.